internal/server/server.go

```diff
@@ -14,28 +15,29 @@ import (
 	newrelic "github.com/newrelic/go-agent"
 	"github.com/newrelic/go-agent/_integrations/nrgrpc"
 	"github.com/odpf/salt/log"
-	"github.com/odpf/salt/server"
+	"github.com/odpf/salt/mux"
 	"github.com/odpf/shield/internal/api"
 	"github.com/odpf/shield/internal/api/v1beta1"
 	"github.com/odpf/shield/internal/server/grpc_interceptors"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/reflection"
 	"google.golang.org/grpc/status"
 )
 
-func registerHandler(ctx context.Context, s *server.MuxServer, gw *server.GRPCGateway, deps api.Deps) {
-	s.RegisterHandler("/admin*", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func registerHandler(ctx context.Context, sh *http.ServeMux, gw *runtime.ServeMux, sg *grpc.Server, deps api.Deps, address string) error {
+	sh.Handle("/admin*", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "pong")
 	}))
 
-	s.RegisterHandler("/admin/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	sh.Handle("/admin/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Why do we need to mount these routes manually? We could pass the `runtime.WithHealthEndpointAt` option in the `runtime.NewServeMux` constructor.

Reply:
Changed this
`[]grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}`

Reply:
`runtime.WithHealthEndpointAt` accepts `grpc_health_v1.HealthClient` which needs `grpc.ClientConnInterface` to initialize. In our current implementation we are NOT making a connection ourself [like compass](https://github.com/odpf/compass/blob/7fc4a8236562d5d73f1253950a9d074c3c61fd82/internal/server/server.go#L111) do. Rather we are using [RegisterShieldServiceHandlerFromEndpoint](https://github.com/odpf/shield/blob/3081952afa0d76e0a4188bf90efee52624d72380/internal/api/v1beta1/v1beta1.go#L28)`.

Should we refactor to make a conn ourselves? Does it have some benefit?

Reply:
Lets refactor, [`RegisterShieldServiceHandlerFromEndpoint`](https://github.com/odpf/shield/blob/3081952afa0d76e0a4188bf90efee52624d72380/internal/api/v1beta1/v1beta1.go#L28) is anyway creating a connection under the hood.

Not needing to mount the ping endpoint is a win and if even if there is some change needed in the response for health checkpoint, using the functionality provided by gRPC gateway's runtime will take care of it. Also, try to go through the implementation for the health endpoint provided by them. That might answer your question w.r.t the benefits.